client/add/add.ctrl.js

```diff
@@ -123,10 +123,27 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
+      .reduce((overall, line) => {
+        if (overall[line.box]) {
+          overall[line.box].push(line);
+        } else {
+          overall[line.box] = [line];
+        }
+        return overall;
+      }, {})
+      .then(lines => {
+        Object.keys(lines).forEach(
+          (line) => line.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE))
+        );
+        return lines;
```

Comment:
I don't think this function is actually doing anything -- neither `forEach` nor `slice` has the effect of mutating `lines`. Also, the `box` variable here is the selected box, but we care about the box this particular group of Pokémon is getting uploaded to.

The goal of this function is to ensure that if a large number of Pokémon are uploaded to a box, the client just fills up the current box page rather than displaying hundreds of uploads and causing lag.

I think we want to do something like this:

```js
import {keyBy} from 'lodash';
import {mapValues} from 'lodash/fp'

const boxesById = keyBy(this.boxes, 'id');

// ...
.then(mapValues((uploadsForBox, boxId) => {
  return uploadsForBox.slice(0, BOX_PAGE_SIZE - (boxesById[boxId].data.contents.length % BOX_PAGE_SIZE))
))
```